### PR TITLE
#M10. Update Hackathon and Profile model with new fields. - Auxfuse

### PR DESCRIPTION
**Pull request by auxfuse**
## Description

Hackathon model updated to include `organisation` and `status` fields.

Profile model updated to include `organisation` field.

- Both `organisation` field entities in each model are set to an initial value of "Code Institute" as per the Issue details, [here](#M10. Update Hackathon and Profile model with new fields. #91). The user can delete the initial value and enter in any organisation name up to a max of 100 characters in length. As is common place with an _optional_ field of this type, I decided it best to leave this field as non-required, allowing a user to submit there profile signup with this field blank, or to simply have a hackathon that might not be based on an organisation, but open to the public with no _official_ capacity ties to same.

- The `status` field incorporates a list, created and separated in a lists.py file added to the hackathon app, and this field is defaulted to "Draft" as per the issue details linked above. The three status types are: 'draft', 'published', 'deleted'. As there is no bespoke *create hackathon* form yet, my only testing on this was via the admin panel to ensure that Draft was the default value and that a user could cycle between the other two.

## Pull request type

- [x] #M10. Update Hackathon and Profile model with new fields. #91 
- [x] Feature & update to existing models

## Testing

Manual Testing via User Registration in Frontend Site and via Admin panel, and explicitely only in Admin for the Hackathon additions as per above.

image

image

image

## Does this introduce a breaking change

- [x] No

### CODE REVIEW

File: accounts/forms.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class ExtendedSignupForm(SignupForm):
     current_lms_module = forms.ChoiceField(
         choices=LMS_MODULES_CHOICES
     )
+    organisation = forms.CharField(max_length=100, initial="Code Institute")
```
Comment (stefdworschak) [inline accounts/forms.py:21]: Could you make this a dropdown field?

File: accounts/models.py [hunk 2]
```diff
@@ -44,18 +44,25 @@ class Profile(models.Model):
         default='',
         choices=LMS_MODULES_CHOICES
     )
+    organisation = models.CharField(
```
Comment (stefdworschak) [inline accounts/models.py:47]: Could you create an extra model for this and change it to a ForeignKey please?

**Review by stefdworschak — CHANGES_REQUESTED** [review]
@auxfuse Looks good overall, I guess the issue did not mention the extra model, so you could not have known about it. Please make the few changes and revert back.

**Comment by stefdworschak** [comment]
Not sure about the extra model's field, but I guess it should have at least `display_name`. If you can think of anything else, please add it.

**Comment by auxfuse** [comment]
Closing this PR as update User object inbound.

**Pull request closed without merging**